golang/cmd/event-ingest/main.go

```diff
@@ -0,0 +1,135 @@
+package main
+
+import (
+	"context"
+	"database/sql"
+	"encoding/json"
+	"fmt"
+	"time"
+
+	"github.com/twmb/franz-go/pkg/kgo"
+)
+
+const batchSize = 1000
+
+type Event struct {
+	OrganizationID int                    `json:"organization_id,omitempty"`
+	CustomerID     *int                   `json:"customer_id,omitempty"`
+	CustID         string                 `json:"cust_id,omitempty"`
+	EventName      string                 `json:"event_name,omitempty"`
+	TimeCreated    time.Time              `json:"time_created,omitempty"`
+	Properties     map[string]interface{} `json:"properties,omitempty"`
+	IdempotencyID  string                 `json:"idempotency_id,omitempty"`
+	InsertedAt     time.Time              `json:"inserted_at,omitempty"`
+}
+
+func main() {
+	seeds := []string{"localhost:9092"}
+	ctx := context.Background()
+
+	// Setup kafka consumer
+	cl, err := kgo.NewClient(
+		kgo.SeedBrokers(seeds...),
+		kgo.ConsumerGroup("default"),
+		kgo.ConsumeTopics("test-topic"),
+		kgo.DisableAutoCommit(),
+	)
+	if err != nil {
+		panic(err)
+	}
+	defer cl.Close()
+
+	//setup db connection
+	db, err := sql.Open("postgres", "postgres://user:password@localhost/dbname?sslmode=disable")
+	if err != nil {
+		panic(err)
+	}
+	defer db.Close()
+
+	// Set up prepared statement for insert
+	stmt, err := db.Prepare(`INSERT INTO events (organization_id, customer_id, cust_id, event_name, time_created, properties, idempotency_id, inserted_at)
+		VALUES ($1, $2, $3, $4, $5, $6, $7, $8)
+		ON CONFLICT DO NOTHING`)
+	if err != nil {
+		panic(err)
+	}
+
+	for {
+		fetches := cl.PollFetches(ctx)
+		if fetches == nil {
+			continue
+		}
+		if fetches.IsClientClosed() {
+			return
```

Comment:
you *probably* want to do `os.Exit(1)` here since the application will be existing with an error. Or - consistent with the rest of the code - do `panic(errors.New("client is closed"))`